features/steps/givens/attributes.py

```diff
@@ -7,12 +7,16 @@
 
 @given("{attribute} = {value}")
 def step_impl(context, attribute, value):
+    pred = operator.eq
     if value == 'empty':
         value = ()
+    elif value == 'not empty':
+        value = ()
+        pred = operator.ne
     else:
         value = ast.literal_eval(value)
```

Comment:
Edge case: what is I wanted the attribute to have the value "empty"? How would this work?

Reply:
In this case it's actually possible. Because that string value would have to be a valid python literal `"empty"` (unwrapped by `ast.literal_eval()`). See comment below, where this edge case is real, because we don't pass around python literals, so can't differentiate between `"empty"` and `empty`. To be addressed later.